src/main/java/com/google/cloud/dataproc/templates/config/QueryConfig.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (C) 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.dataproc.templates.config;
+
+import com.google.common.base.MoreObjects;
+import jakarta.validation.constraints.NotBlank;
+
+public class QueryConfig {
+
+  @NotBlank private String sql;
```

Comment:
It would be nice to have a way to ask spark if this is valid sql before we start the spark session so we can fail faster, but I don't think there's an easy way to do that.

Reply:
Agree that this is desirable, but not urgent. we could create a FR to not loose track of this